route/build/src/write_restart_pio.f90

```diff
@@ -440,11 +452,39 @@ SUBROUTINE define_IRFbas_state(ierr, message1)
      call def_var(pioFileDesc, meta_irf_bas(iVar)%varName, meta_irf_bas(iVar)%varType, ierr, cmessage, &
                   pioDimId=dim_IRFbas, vdesc=meta_irf_bas(iVar)%varDesc, vunit=meta_irf_bas(iVar)%varUnit)
      if(ierr/=0)then; message1=trim(message1)//trim(cmessage); return; endif
-
    end do
 
   END SUBROUTINE define_IRFbas_state
 
+  SUBROUTINE define_bas_tracer_state(ierr, message1)
```

Comment:
Needs a description.

Reply:
subroutine name is changed to `write_bas_solute_state`.  The same question as reading state, wondering how much descriptions are needed here. This and a bunch of other same routines in state writing module are writing states (in restarts file). Do you think more detailed descriptions are needed here and the other routines?

Reply:
I added short, general description for `define_state_nc`